tests/core/string/test_fuzzy_search.h

```diff
@@ -0,0 +1,157 @@
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_FUZZY_SEARCH_H
+#define TEST_FUZZY_SEARCH_H
+
+#include "core/string/fuzzy_search.h"
+#include "tests/test_macros.h"
+#include <chrono>
+#include <iostream>
+
+namespace TestFuzzySearch {
+
+struct FuzzySearchTestCase {
+	String query;
+	String expected;
+};
+
+struct FuzzySearchTestOutcome {
+	String top_result;
+	int result_count;
+};
+
+struct FuzzySearchBenchmarkResult {
+	double average_ms;
+	double std_dev_ms;
+	FuzzySearchTestOutcome outcome;
+};
+
+// Ideally each of these test queries should represent a different aspect, and potentially bottleneck, of the search process.
+const FuzzySearchTestCase test_cases[] = {
+	// Short query, many matches, few adjacent characters
+	{ "///gd", "./menu/hud/hud.gd" },
+	// Filename match with typo
+	{ "sm.png", "./entity/blood_sword/sam.png" },
+	// Multipart filename word matches
+	{ "ham ", "./entity/game_trap/ha_missed_me.wav" },
+	// Single word token matches
+	{ "push background", "./entity/background_zone1/background/push.png" },
+	// Long token matches
+	{ "background_freighter background png", "./entity/background_freighter/background/background.png" },
+	// Many matches, many short tokens
+	{ "menu menu characters wav", "./menu/menu/characters/smoker/0.wav" },
+	// Maximize total matches
+	{ "entity gd", "./entity/entity_man.gd" }
+};
+
+double calculate_mean(const Vector<double> &p_numbers) {
+	double sum = 0.0;
+	for (double num : p_numbers) {
+		sum += num;
+	}
+	return sum / static_cast<double>(p_numbers.size());
+}
+
+double calculate_std_dev(const Vector<double> &p_numbers) {
+	double mean = calculate_mean(p_numbers);
+	double variance = 0.0;
+	for (double num : p_numbers) {
+		variance += (num - mean) * (num - mean);
+	}
+	variance /= static_cast<double>(p_numbers.size());
+	return std::sqrt(variance);
+}
+
+auto load_test_data(int p_repeat = 1) {
```

Comment:
```suggestion
Vector<String> load_test_data(int p_repeat = 1) {
```
`auto` is [not allowed](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword)